src/utils/cid/mod.rs

```diff
@@ -31,37 +31,54 @@ impl CidCborExt for Cid {}
 
 pub const BLAKE2B256_SIZE: usize = 32;
 
-/// `CidVariant` is an enumeration of known CID types that are used in the Filecoin blockchain. CIDs
+/// `SmallCid` encapsulates an enumeration of known CID types that are used in the Filecoin blockchain. CIDs
 /// contain a significant amount of static data (such as version, codec, hash identifier, hash
 /// length). This static data represented by a single tag in the `enum`.
 ///
 /// Nearly all Filecoin CIDs are `V1`,`DagCbor` encoded, and hashed with `Blake2b256` (which has a hash
-/// length of 256 bits). Naively representing such a CID requires 96 bytes but `CidVariant` does it in
-/// only 40 bytes. If other types of CID become popular, they can be added to the `CidVariant`
+/// length of 256 bits). Naively representing such a CID requires 96 bytes but `SmallCid` does it in
+/// only 40 bytes. If other types of CID become popular, they can be added to the `SmallCid`
 /// structure.
 ///
 /// The `Generic` variant is used for CIDs that do not fit into the other variants.
 /// These variants are used for optimizing storage of CIDs in the `FrozenCids` structure.
 #[cfg_attr(test, derive(derive_quickcheck_arbitrary::Arbitrary))]
 #[derive(Clone, Debug, PartialEq, Eq, Hash)]
-pub enum CidVariant {
+pub struct SmallCid(SmallCidInner);
+
+impl SmallCid {
+    pub fn cid(&self) -> Cid {
+        match &self.0 {
+            SmallCidInner::Generic(cid) => **cid,
+            SmallCidInner::V1DagCborBlake2b(digest) => Cid::new_v1(
+                DAG_CBOR,
+                multihash::Multihash::wrap(Blake2b256.into(), digest)
+                    .expect("failed to convert Blake2b digest to V1 DAG-CBOR Blake2b CID"),
```

Comment:
Isn't it a failure in creating a MultiHash rather than creating a Cid?

Reply:
Yes, but because this is just a piece inside the process of ultimately creating a CID, I thought this error message would be more descriptive. I'll add a little clarification though.